zebra-network/src/peer_set/initialize.rs

```diff
@@ -125,21 +119,16 @@ where
         let _ = demand_tx.try_send(());
     }
 
-    tokio::spawn(
-        crawl_and_dial(
-            config.new_peer_interval,
-            demand_tx,
-            demand_rx,
-            candidates,
-            connector,
-            peerset_tx,
-        )
-        .map(|result| {
-            if let Err(e) = result {
-                error!(%e);
-            }
-        }),
-    );
+    let _crawl_guard = tokio::spawn(crawl_and_dial(
+        config.new_peer_interval,
+        demand_tx,
+        demand_rx,
+        candidates,
+        connector,
+        peerset_tx,
+    ));
+    // TODO(jlusby): impl `DerefMut` for tower::Buffer and uncomment this
+    // peer_set.push_join_handle(crawl_guard);
```

Comment:
Need to find some way to fix this, RN theres a circular dependency between the `crawl_and_dial` task and the `peer_set`, either we need some way to be able to hold onto a reference to the guards in the peer_set while its already in the Buffer (`Arc`?) or we need a way to access the peer_set while it's in the `Buffer` (`DerefMut`?).